qiskit_dynamics/signals/signals.py

```diff
@@ -801,15 +799,15 @@ def __init__(self, signal_list: List[Signal]):
 
         super().__init__(signal_list)
 
-        self._eval_complex_value = lambda t: [sig.complex_value(t) for sig in self.components]
-        self._eval_signals = lambda t: [sig(t) for sig in self.components]
+        self._eval_complex_value = lambda t: unp.asarray([sig.complex_value(t) for sig in self.components])
+        self._eval_signals = lambda t: unp.asarray([sig(t) for sig in self.components])
 
-    def complex_value(self, t: Union[float, np.ndarray, jnp.ndarray]) -> Union[np.ndarray, jnp.ndarray]:
```

Comment:
An error occurs in the case that we cannot use jax if we use a type hint of something relating to jax.